File: groupAnagrams/groupAnagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        std::unordered_map<std::string,std::vector<std::string>> res;

        for (const auto&  s : strs){
            std::vector<int> freqs(26,0);
            for (char c : s){
                freqs[c - 'a']++; // Create the freqeuency map
            }

            std::string key = to_string(freqs[0]);
            for (int i = 1; i < 26; i++){
                key += ',' + to_string(freqs[i]); // Create a unique key for the frequency map
            }

            res[key].push_back(s);
        }

        std::vector<std::vector<std::string>> result;
        for (const auto& pair : res){
            result.push_back(pair.second); // Append the list/vector of anagram strings from each key to the results vector

        }

        return result;
    }
};

File: groupAnagrams/groupAnagrams.py
from collections import defaultdict

class Solution:
    def groupAnagrams(self, strs: list[str]) -> list[list[str]]:
        res = defaultdict(list) # Create a dictionay

        for s in strs:
            freqs = [0] * 26
            for c in s:
                freqs[ord(c) - ord('a')] += 1 # Create a frequency map
            res[tuple(freqs)].append(s) # Add the original word to the frequency map being used as a key in the dict

        return list(res.values()) # Since all the values are lists of anagrams we return them

        